backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/service/SpecificationDataService.kt

```diff
@@ -74,6 +75,8 @@ class SpecificationDataService @Autowired constructor(
     fun extractId(json: String): String? =
         try {
             JsonPath.read<String>(json, "$.info.x-api-id")
+        } catch (exception: ClassCastException) {
+            JsonPath.read<Integer>(json, "$.info.x-api-id").toString()
```

Comment:
If the value is an `Integer`, can't that also be read as a `String`? Also, what happens if it's neither a `String` nor an `Integer`?

Reply:
No, without this catch block, the upload fails. And the type parameter to `JsonPath.read` is required.</br></br>The JSON grammar, being Javascript, only offers "string" and "number" -- However I just tried with a non-integer number, and realised that will also cause an exception / failed uploaded because `java.lang.Double` cannot be cast to either of these